lib/generators/administrate/routes/routes_generator.rb

```diff
@@ -50,30 +53,92 @@ def dashboard_resources
         end
       end
 
+      def dashboard_resources_hash
+        output = [{}, []]
+        dashboard_resources.each do |resource_path|
+          split_path = resource_path.split("/")
+          output = [
+            add_namespace_to_hash(split_path, output[0]),
+            add_resources_to_hash(split_path, output[1]),
+          ]
+        end
+        output
+      end
+
+      def add_namespace_to_hash(split_path, current_state)
+        if split_path.length > 1
+          current_state[split_path[0]] ||= [{}, []]
+          current_namespace = current_state[split_path[0]]
+          current_namespace = [
```

Comment:
used to fix a line length issue..

Reply:
It is used in the declaration, lines 73 and 74

Reply:
Yes and no:

* Line 71: first definition of `current_namespace`.
* Lines 72-75: second definition, referring back to the previous one.
* Line 76: end of `if` block.
* Line 77: last expression of the method, which only refers to `current_state`.
* Line 78: end of method.

So after its last definition (lines 72-75), `current_namespace` is never used for anything. The method doesn't do anything afterwards that makes use of it, and I don't think it takes part in any side-effect.

So this method is equivalent to the following:

```ruby
def add_namespace_to_hash(split_path, current_state)
  if split_path.length > 1
    current_state[split_path[0]] ||= [{}, []]
  end
  current_state
end
```

Reply:
73, 74. add to the `current namespace` which is current done by passing in the `current namespace element` to be added to . (I can see why the linter is unhappy, but not sure if the process fits what the linter is unhappy about) I only put in the `current namespace` variable to shorting the line length done - possibly there is a better way to do this

Reply:
OK, you are right because the specs fail without those lines :sweat_smile: 

I'm starting to see... The assignment, the array, and the first of the two calls to `add_namespace_to_hash` are unnecessary, right? I think they are red herrings and were throwing me off:

```ruby
      def add_namespace_to_hash(split_path, current_state)
        if split_path.length > 1
           current_namespace = current_state[split_path[0]]
           add_resources_to_hash(split_path[1..], current_namespace[1])
        end
        current_state
      end
```

Reply:
so the `add_namespace_to_hash` deals with the next level of nested resources while the `add_resources_to_hash` adds resources into the current namespace level.

Reply:
I've not thought about whether the methods need to be called as assignment in the array positions... (so may well work without that) I do think it needs to self call the `add_namespace_to_hash` though to capture the nested namespaces